closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -26,13 +26,78 @@
 #include <openssl/aes.h>
 #include <openssl/err.h>
 
+#include <assert.h>
 #include <jni.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <assert.h>
 
 #include "jdk_crypto_jniprovider_NativeCrypto.h"
+#include "NativeCrypto_md.h"
+
+//Type definitions of function pointers
+typedef char * OSSL_error_string_t(unsigned long, char *);
```

Comment:
These are technically, *not* function pointers, if you want a function pointer you should write:
> typedef char * (*OSSL_error_string_t)(unsigned long, char *);

and replace `OSSL_error_string_t *` with `OSSL_error_string_t`.